src/ADP.Portal.Core/Git/Services/FluxTeamConfigService.cs

```diff
@@ -0,0 +1,520 @@
+                                finalFiles.TryGetValue(key, out var existingEnv))
+                            {
+                                ((List<object>)existingEnv[Constants.Flux.RESOURCES_KEY]).Add($"../../{service.Name}");
+                            }
+                            else
+                            {
+                                var newFile = template.Value.DeepCopy();
+                                if (template.Key.Equals(Constants.Flux.TEAM_ENV_KUSTOMIZATION_FILE, StringComparison.InvariantCultureIgnoreCase))
+                                {
+                                    ((List<object>)newFile[Constants.Flux.RESOURCES_KEY]).Add($"../../{service.Name}");
+                                }
+
+                                var tokens = new List<FluxConfig>
+                            {
+                                new() { Key = Constants.Flux.TEMPLATE_VAR_VERSION, Value = Constants.Flux.TEMPLATE_VAR_DEFAULT_VERSION },
+                                new() { Key = Constants.Flux.TEMPLATE_VAR_VERSION_TAG, Value = Constants.Flux.TEMPLATE_VAR_DEFAULT_VERSION_TAG },
+                                new() { Key = Constants.Flux.TEMPLATE_VAR_MIGRATION_VERSION, Value = Constants.Flux.TEMPLATE_VAR_DEFAULT_MIGRATION_VERSION },
+                                new() { Key = Constants.Flux.TEMPLATE_VAR_MIGRATION_VERSION_TAG, Value = Constants.Flux.TEMPLATE_VAR_DEFAULT_MIGRATION_VERSION_TAG },
+                                new() { Key = Constants.Flux.TEMPLATE_VAR_PS_EXEC_VERSION, Value = Constants.Flux.TEMPLATE_VAR_PS_EXEC_DEFAULT_VERSION }
+                            };
+                                tokens.ForEach(newFile.ReplaceToken);
+
+                                tokens =
+                                [
+                                    new() { Key = Constants.Flux.TEMPLATE_VAR_ENVIRONMENT, Value = environment.Name[..3]},
+                                new() { Key = Constants.Flux.TEMPLATE_VAR_ENV_INSTANCE, Value = environment.Name[3..]},
+                            ];
+                                var tenantConfigVariables = tenantConfig.Environments.First(x => x.Name.Equals(environment.Name)).ConfigVariables ?? [];
+
+                                tokens.Union(environment.ConfigVariables).Union(tenantConfigVariables).ForEach(newFile.ReplaceToken);
+                                finalFiles.Add(key, newFile);
+                            }
+                        });
+                }
+            }
+            return finalFiles;
+        }
+
+        private static Dictionary<object, object> UpdateServiceKustomizationFiles(Dictionary<object, object> serviceTemplate, string templateKey, FluxService service)
+        {
+            if (templateKey.Equals(Constants.Flux.TEAM_SERVICE_KUSTOMIZATION_FILE) && service.HasDatastore())
+            {
+                ((List<object>)serviceTemplate[Constants.Flux.RESOURCES_KEY]).Add("pre-deploy-kustomize.yaml");
```

Comment:
Is there no way this could be done using a dto with the correct property types, rather than casting elements of a Dictionary<object, object>?

Reply:
@ajaysoni-cap ?

Reply:
We will have to create too many DTOs and maintain it as per the Flux framework update so we decided to use the objects to keep it generic.

Reply:
My concern is if the template is misconfigured, or the `RESOURCES_KEY` section is missing. this would throw an error if it isnt a `List<object>` or is missing the key. If it has to stay looslely typed with `Dictionary<object, object>`, then we should do more checks here to make sure it is what we expect rather than just assuming.